models/web/scratch/snowplow_media_player_base_this_run.sql

```diff
@@ -64,21 +76,35 @@ with prep as (
     inner join {{ ref("snowplow_media_player_interactions_this_run") }} i
     on i.play_id = d.play_id
 
-    where i.event_type in ('percentprogress') and (i.start_tstamp <= d.first_seek_time or d.first_seek_time is null)
+    where i.percent_progress is not null and (i.start_tstamp <= d.first_seek_time or d.first_seek_time is null)
```

Comment:
This is changed because `percent_progress` could still be null when `event_type` is `percentprogress`?

Reply:
It changed because the previous solution was incomplete, not including 100 which comes from the `ended` event. (It was changed from NULL to 100 within the percent_progress field by the model itself during the interactions_this_run to account for 100% automatically). So I picked the wrong field originally. The other solution could have been to include ('percentprogress', 'ended') but I found it cleaner this way cause then I'd have had to put a case when statement as well.